pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -374,17 +379,17 @@ func (c *Controller) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 				}
 				return nil
 			}).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(o.Shoot.HibernationEnabled),
-			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, waitUntilOperatingSystemConfigReady),
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, initializeShootClients, waitUntilOperatingSystemConfigReady, deployKubeScheduler),
 		})
 		_ = g.Add(flow.Task{
 			Name:         "Deploying metrics-server system component",
 			Fn:           flow.TaskFn(botanist.DeployMetricsServer).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(o.Shoot.HibernationEnabled),
-			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, waitUntilOperatingSystemConfigReady),
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, waitUntilOperatingSystemConfigReady, deployKubeScheduler),
 		})
 		deployManagedResourcesForAddons = g.Add(flow.Task{
 			Name:         "Deploying managed resources for system components and optional addons",
 			Fn:           flow.TaskFn(botanist.DeployManagedResourceForAddons).RetryUntilTimeout(defaultInterval, defaultTimeout).SkipIf(o.Shoot.HibernationEnabled),
-			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, ensureShootClusterIdentity),
+			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager, initializeShootClients, ensureShootClusterIdentity, deployKubeScheduler),
```

Comment:
We now deploy the `kube-scheduler` only after `gardener-resource-manager` is ready to make sure that GRM is up and running and able to reconcile the shoot access secret for `kube-scheduler`. If we deployed `kube-scheduler` earlier (while GRM is not yet ready) then it would lead to a couple of crashing pod starts since its token was not yet generated by GRM.

As for the changes w.r.t. CoreDNS, etc.: Introducing the task dependency to `deployKubeScheduler` is not strictly needed (we can create the `Deployment`s, `DaemonSet`s, etc. for all the system components in the shoot even without a `kube-scheduler`). However, from a logical POV it makes more sense to only start deploying workload into the shoot after the `kube-scheduler` got deployed.